### PR TITLE
Remove ppx as a runtime link dependency

### LINKED ISSUE #80: Linking vchan adds 5 MB to my unikernel
qubes-mirage-firewall depends on vchan, which depends on ppx_sexp_conv, which requires compiler-libs.common, which increases the size of my unikernel by about 5 MB.

Removing `ppx_sexp_conv` from vchan's `META` file seems to solve the problem, so I assume it's not necessary.

### PR DESCRIPTION
**Pull request by djs55**
Remove ppx as a runtime link dependency

This is ported from @avsm's patch in #81

Fixes #80 
Obsoletes #81

### CODE REVIEW

File: _oasis [hunk 1]
```diff
@@ -26,6 +26,8 @@ Library vchan
   BuildDepends:       lwt, cstruct (>= 1.9.0), io-page, cstruct.ppx,
                       mirage-types (>= 1.1.3), mirage-types.lwt,
                       xenstore.client, sexplib, ppx_deriving, ppx_sexp_conv
+  XMETARequires:      lwt, cstruct, io-page, mirage-types, mirage-types.lwt,
+                      xenstore.client, sexplib, result
```
Comment (talex5) [inline _oasis]: Why does this add `result` now? It isn't mentioned elsewhere as a dependency.

**Comment by talex5** [comment]
Might want to take the changelog entry too?

File: _oasis (hunk 1 above)
Comment (djs55) [inline _oasis]: Ah sorry `result` is added by #82 -- I'll probably have to rebase this anyway

**Comment by djs55** [comment]
My cunning plan is to merge as many PRs as possible and then update the changelog and release in a batch.

**Comment by talex5** [comment]
Good plan - that will simplify the merge indeed :-)

**Pull request merged**